### DISCUSSION TITLE
Only allow downloading in supabase storage

### DISCUSSION THREAD

**Discussion by RealCyGuy** [supabase/supabase#1664] [Questions] [answered]
Is it possible to not allow all `SELECT` permissions, only download?

**Comment by inian** [comment, marked as answer]
Hey @RealCyGuy, can you share the problem you are trying to solve? If the user is able to download the file, there is nothing special about reading that row in the database too. Are you trying to hide fields like `inserted_at` or `owner` or considering something else?

**Reply by RealCyGuy** [reply]
I want it to be only allowed to download but not view the whole list of files.

**Reply by inian** [reply]
Hey @RealCyGuy, we have [launched public buckets](https://supabase.io/docs/guides/storage#accessing-objects) now. You should be able to make your bucket public without allowing users to enumerate the bucket.
